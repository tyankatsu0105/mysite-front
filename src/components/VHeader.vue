<template>
  <header class="VHeader">
    <button 
      class="VHeader_TriggerButton" 
      type="button" 
      @click="isOpen = !isOpen">
      Menu
    </button>
    <transition name="header">
      <div 
        v-if="isOpen" 
        class="VHeader_NavContainer">
        <nav 
          class="VHeader_Nav">
          <button 
            class="VHeader_NavClose" 
            @click="isOpen = !isOpen"><font-awesome-icon icon="times" />
          </button>
          <ul class="VHeader_NavList">
            <li class="VHeader_NavListItem">
              <g-link 
                :to="{ name: 'home' }" 
                class="VHeader_NavListItemLink"
                exact>Home</g-link>
            </li>
            <li class="VHeader_NavListItem">
              <g-link 
                :to="{ name: 'about' }" 
                class="VHeader_NavListItemLink"
                exact>About</g-link>
            </li>
          </ul>
        </nav>
      </div>
      
    </transition>
  </header>
</template>

<script>
export default {
  data() {
    return {
      isOpen: false
    };
  }
};
</script>

<style lang="scss" scoped>
@import "@/styles/utility/transition-header.scss";
.VHeader {
  position: fixed;
  top: 20px;
  left: 20px;
  z-index: 1;

  &_TriggerButton {
    position: relative;
    padding-bottom: 8px;
    font-size: 1.2rem;
    &::before {
      position: absolute;
      bottom: 0;
      left: 0;
      width: 100%;
      height: 2px;
      content: "";
      background-color: $color-accent;
      transition: $tst-default;
    }

    &:hover {
      &::before {
        width: calc(100% + 20px);
      }
    }
  }

  &_Nav {
    position: relative;
    padding: 36px 0;
  }
  &_NavClose {
    position: absolute;
    top: 5px;
    right: 5px;
    font-size: 1.5rem;
    color: inherit;
  }

  &_NavContainer {
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    color: $color-primary;
    background-color: #fff;
  }
  &_NavListItemLink {
    padding: 8px 40px;

    &.active {
      background-color: $color-font;
    }
  }
}
</style>
