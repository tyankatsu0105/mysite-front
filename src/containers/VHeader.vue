<template>
	<header class="VHeader">
		<HeaderOpenButton @open-header="changeHeaderStatus" />
		<transition
			name="header"
		>
			<HeaderMenu
				v-if="isOpenHeader"
				@close-header="changeHeaderStatus"
			/>
		</transition>
	</header>
</template>

<script>
import HeaderOpenButton from "@/basics/HeaderOpenButton.vue";
import HeaderMenu from "@/components/HeaderMenu.vue";
export default {
  components: {
    HeaderOpenButton,
    HeaderMenu
  },
  data() {
    return {
      isOpenHeader: false
    };
  },
  methods: {
    changeHeaderStatus(status) {
      this.isOpenHeader = status;
    }
  }
};
</script>

<style lang="scss" scoped>
@import "@/styles/utility/transition-header.scss";
.VHeader {
  position: fixed;
  top: 20px;
  left: 20px;
  z-index: 1;

  &::after {
    position: fixed;
    top: 0;
    left: 0;
    z-index: -1;
    width: 0;
    height: 0;
    content: "";
    border-color: transparentize($color-primary, 0.1) transparent transparent
      transparent;
    border-style: solid;
    border-width: 120px 120px 0 0;
  }
}
</style>
