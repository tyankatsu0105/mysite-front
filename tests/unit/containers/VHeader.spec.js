import { shallowMount } from "@vue/test-utils";
import VHeader from "@/containers/VHeader.vue";

describe("VHeader", () => {
  it("render", () => {
    const wrapper = shallowMount(VHeader);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
