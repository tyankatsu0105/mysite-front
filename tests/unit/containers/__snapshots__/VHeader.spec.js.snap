// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHeader render 1`] = `
<header class="VHeader">
  <headeropenbutton-stub></headeropenbutton-stub>
  <!---->
</header>
`;
